src/main/java/net/glowstone/util/PathFinder.java

```diff
@@ -0,0 +1,231 @@
+package net.glowstone.util;
+
+import org.bukkit.util.Vector;
+
+import java.util.*;
+
+/**
+ * General class for path finding.
+ * TODO: Implement other path finding algorithms (BFS, DFS, D*, potentially more)
+ */
+
+public class PathFinder {
+
+	private final Vector startNode;
+	private final Vector endNode;
+	private List<Vector> path;
+	private Map<Vector, Vector> fromPath;
+
+
+	/**
+	 * Creates a new PathFinder object.
+	 *
+	 * @param startNode Starting position
+	 * @param endNode Ending position
+	 */
+	public PathFinder(final Vector startNode, final Vector endNode) {
+		// Save our start and end points
+		this.startNode = startNode;
+		this.endNode = endNode;
+		path = new ArrayList<Vector>();
+	}
+
+
+	/**
+	 * Gets the start (initial) node
+	 * @return Vector object
+	 */
+	public Vector getStartNode() {
+		return startNode;
+	}
+
+
+	/**
+	 * Gets the end (goal) node.
+	 * @return Vector object
+	 */
+	public Vector getEndNode() {
+		return endNode;
+	}
+
+	/**
+	 * Gets the path once it has been constructed.
+	 *
+	 * @note This method should only be called AFTER AStarSearch has been called
+	 * @return List of Vector objects in order from start to end
+	 */
+	public List<Vector> getPath() {
+		return path;
+	}
+
+
+	/**
+	 * Tries to find a path between the start and end points. This method currently
+	 * DOES NOT handle cases where a path could not potentially exist.
+	 */
+	public void AStarSearch() {
+		Queue<Vector> frontier = new PriorityQueue<>();
+		fromPath = new HashMap<>();
```

Comment:
Add `this` before.

Reply:
Do you want me to do this for all member variables?